.cloudbuild/library_generation/library_generation.Dockerfile

```diff
@@ -35,6 +35,7 @@ SHELL [ "/bin/bash", "-c" ]
 ARG OWLBOT_CLI_COMMITTISH=ac84fa5c423a0069bbce3d2d869c9730c8fdf550
 ARG PROTOC_VERSION=25.5
 ARG GRPC_VERSION=1.67.1
+ARG JAVA_FORMAT_VERSION=1.7
```

Comment:
I think we can upgrade the formatter, the latest version is 1.23.0.

It will cause code formatting so it's better in a separate PR.

Reply:
This fomatter is not part of client library and we should be able use a new version of Java to run it.

Reply:
Since it is a jar, I think it still needs to be run with Java. I see that we are currently using [Java 21](https://github.com/googleapis/sdk-platform-java/blob/main/.cloudbuild/library_generation/library_generation.Dockerfile#L17) in our build file, but we probably  want to use a lower version to guarantee the compatibility since our client library still supports Java 8.

Reply:
I think this repo has a check to verify the compiled code is compatible with Java 8 (maven profile should make sure of that). Of course, we can change it to Java8 as well.

On the other hand, running generator/formatter jar only requires JRE which I think we can choose a different version from JDK. We can optimize the image to only include a JRE but not a JDK.